client/src/components/Reviews/ReviewMapper.jsx

```diff
@@ -1,12 +1,35 @@
-// import React, { useState } from 'react';
 import React from 'react';
 import Review from './Review.jsx';
 
-const ReviewMapper = ({ reviewList, StarReview }) => {
+
+const ReviewMapper = ({ reviewList, StarReview, filterRating, currentSearch }) => {
+
+  //mapper with filterRating and currentSearch filters applied
   return (
-    <div>{reviewList.map(review =>
-      <Review review={review} key={reviewList.indexOf(review)} StarReview = {StarReview} />)}</div>
+    <div>{reviewList.map(review => {
+      if (filterRating.includes(review.rating) && currentSearch && review.summary.toLowerCase().includes(currentSearch.toLowerCase())) {
+        return <Review review={review} key={reviewList.indexOf(review)} StarReview={StarReview} />;
+      } else if (filterRating.includes(review.rating && !currentSearch)) {
+        return <Review review={review} key={reviewList.indexOf(review)} StarReview={StarReview} />;
+      } else if (filterRating.length === 0 && currentSearch && review.summary.toLowerCase().includes(currentSearch.toLowerCase())) {
+        return <Review review={review} key={reviewList.indexOf(review)} StarReview={StarReview} />;
+      } else if (filterRating.length === 0 && !currentSearch) {
+        return <Review review={review} key={reviewList.indexOf(review)} StarReview={StarReview} />;
```

Comment:
these else if's look a little redundant / all return the same thing?

Reply:
tried || approach, but that seemed to mess up the functionality and the filters would no longer work properly. Will consider other alternatives when refactoring.